### ISSUE TITLE
Incorrect denominator in bfmi

### ISSUE THREAD

**Issue by sethaxen** [arviz-devs/arviz#858]
The original equation for `bfmi` is at the bottom of page 7 in [this paper](https://arxiv.org/pdf/1604.00695v1.pdf). The numerator in numpy for a vector `E` is `(length(E) - 1) * np.mean(np.square(np.diff(E)))`. The denominator is `(length(E) - 1) * np.var(E, ddof = 1)`. The two scaling factors cancel and we get
```python
def bfmi(E):
    return np.mean(np.square(np.diff(E))) / np.var(E, ddof = 1)
```
The current implementation uses `ddof=0`, so it's off from the equation in the paper by `length(E) / (length(E) - 1)`

**Additional context**
PyStan's implementation [here](https://github.com/stan-dev/pystan/blob/cbc5c0fc8597504776c08a138adc9ae63329e6a1/pystan/diagnostics.py#L213-L217) is equivalent to the paper's. RStan's implementation of `get_bfmi` [here](https://rdrr.io/cran/rstan/src/R/check_hmc_diagnostics.R) has a different implementation than either ArviZ's current one or PyStan's. Theirs amounts to the original paper's equation multiplied by `(length(E) - 1) / length(E)`.

**Comment by nitishp25** [comment]
Hi, can I work on this issue?

**Comment by sethaxen** [comment]
That would be nice! There's was an offline discussion that it would be good to check with the Rstan team and post on Stan's discourse to see if there's a reason for the discrepancy from the paper. I filed https://github.com/stan-dev/rstan/issues/712, but that's as far as I got.

**Comment by nitishp25** [comment]
I've fixed this in PR #991. Ready to merge :)
